### DISCUSSION TITLE
Code generation for function call (using lookup variable)

### DISCUSSION THREAD

**Discussion by efad2022** [climateinteractive/SDEverywhere#262] [Q&A] [answered]
I have an .mdl with 
> csvtest=
>     GET DIRECT LOOKUPS( 'test.csv' , ',' , 'A' , 'B2' )
>     ~
>     ~        |
>
> DataE=
>     csvtest(timestamp)

Note: timestamp is a variable which depends on Time

The code to call csvtest is not generated correctly unless I use subscripts.
I guess the issue is in the grammar:
expr:   Id '(' exprList? ')'              # Call
      |   Id ('[' subscriptList ']')? '(' expr ')' # LookupCall
      
One idea I have in mind is to check if the function name is a variable name (in [equation_reader.js #232](https://github.com/climateinteractive/SDEverywhere/blob/260a8721b9d72f416801c4992f4b2f046254936f/packages/compile/src/model/equation-reader.js#L230)). Then I could emit a lookup (similar to the LookupCall implementation).
I'm not sure if antlr4 parser provides me with variables context.

Does anyone have a better or easier idea for this case?

**Comment by efad2022** [comment]
As a workaround I have created a macro for this

**Comment by chrispcampbell** [comment, marked as answer]
@efad2022: Thanks for submitting this.  It does look like a bug in the compiler, so I filed a new issue #263 with additional comments and assigned it to @ToddFincannonEI since he should have a good idea of how best to address it.
